src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/MagicalZombiePills.java

```diff
@@ -47,7 +51,7 @@ public EntityInteractHandler getItemHandler() {
         return (e, item, offhand) -> {
             Entity entity = e.getRightClicked();
 
-            if (entity.getType() == EntityType.ZOMBIE_VILLAGER) {
+            if (entity instanceof ZombieVillager || entity instanceof PigZombie) {
```

Comment:
Also, why are we using PigZombie instead of Piglin? Then the version check should be a bit different.
Is a piglin entity really an instance of PigZombie?

Reply:
Because we want to turn a PigZombie into a Piglin, not vice versa

Reply:
But in 1.16 there is no PigZombie. Or am I wrong?

Reply:
We are looking for zombified piglins, right?

Reply:
Zombie pigman were renamed to Zombified Piglin, theyre the same

Reply:
I believe you. Did you test this ingame?

Reply:
Yep, Spigot still calls the Zombified Pigling PigZombie internally. That's the corrret class

Reply:
Okay then